### ISSUE TITLE
WSOD in flight mode

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#5216]
## Issue Overview
I hope you see the funny side of this problem. I'm sure there is an easy fix. 
But I also believe that the current solution needs some form of documentation associated with the use of unpkg.com and rawgit.com to load JavaScript files. 

This problem happened at the worst possible time! 
Halfway through my Gutenberg presentation at WordPress Portsmouth Meetup last night. 

I'd already had major problems getting my laptop connected to the projector - the HDMI connector wasn't working at all and the VGA connector misbehaved badly. We were already behind schedule and then this happened....

I'd given an introduction on some slides and had switched to the browser, where I showed a page I'd previously created with an analogy of progression in the music industry. 
- From master tape, to vinyl, 
- the CD version 
- and now the ability to stream music directly from a website. 
- The page also contained a WordPress embed to another of my blogs. 

So now I was going to show the blocks. 
- I clicked on Edit
- We waited an interminably long time.
- The screen cleared and
- Nothing

Panic. Problem determination time. The console was full of messages. Failure to load files from unpkg.com and associated horribleness. Something that @azaozz had mentioned a few days earlier. 
I thought it might have been due to my plugin - which I'd been using to test fixes for the WSOD in the Classic editor. So I deactivated that and tried again. It was failing in multiple environments. Then after quite a few other false starts, it dawned on us that the real cause of the problem was what we'd seen in the console. I was not connected to the network. A short while later we were back in action. 

This morning I reproduced the problem. 
My normal development environment is a Windows machine with a hand  cranked Apache and PHP switchable between 7.1 / 7.2.
I was running the latest development build.
I'm normally connected to a Wireless network,
But often develop and test code quite happily while disconnected. 

## Steps to Reproduce (for bugs)

1. Go to wp-content/plugins/gutenberg/vendor and touch the react.hhhhhhhh.js file to be over 24 hours old
2. Switch to Flight mode or otherwise disconnect from the network
3. Browse a page - notice some messages in the console
4. Clear the console
5. Edit the page
6. More messages in the console and WSOD.

## Expected Behavior

The Editor appears (quick as a flash) and the demo continues.

## Current Behavior

Under the covers gutenberg_register_vendor_script() has a little trouble updating the cached files which it has determined to be out of date,  but knowingly enqueues the external, inaccessible URL.

The comment says
```
			// The request failed; just enqueue the script directly from the
			// URL.  This will probably fail too, but surfacing the error to
			// the browser is probably the best we can do.
```

## Possible Solution
If it ain't broke don't fix it. 
i.e. If there's a failure to load a new file then assume the cached file is still OK to use. 

In my scenario we had to wait for cache rebuild to fail 11 times!
Perhaps we should short circuit subsequent server requests. 

## Screenshots / Video

![issue-wsod](https://user-images.githubusercontent.com/2474435/36595058-ab6bef36-1898-11e8-9f0f-a30cfeb7f09d.gif)

## Todos
- [ ] Tests
- [ ] Documentation

**Comment by aduth** [comment]
Considering as fixed by #5499.

Let me know if there is remaining work.

**Comment by aduth** [comment]
Loading resources from unpkg.com is really just a convenience for development. It's expected as part of the merge process that these dependencies would be treated in much the same way as existing dependencies in core, with a downloaded copy at a fixed version, updated intentionally when a version bump is decided upon.
